### ISSUE TITLE
"Retrying in background" Setup loop

### ISSUE THREAD

**Issue by gnieboer** [travisghansen/hass-pfsense#68]
Initial setup, pfSense 2.5.0 and hass-pfsense 0.4.1, configured for 5 max sessions.

HA shows "retrying setup" and pfSense shows:
192.168.0.100 - homeassistant@mydomain.local [21/Mar/2022:18:36:30 -0400] "POST /xmlrpc.php HTTP/1.1" 200 170 "-" "Python-xmlrpc/3.9"
Basically 5 identical requests in sequence every 10 seconds.

My firewall leverages AD authentication, and I created a specific user in AD, gave it sufficient pfSense permissions, and put in username@mydomain.local with the password in the setup.  It reported success, but then appears to be "retrying setup" for the last 30+ minutes.

Otherwise on the firewall side, I'm running several VLANs and have 2 outgoing VPN circuits, and running several add on packages including suricata, zabbix, pimd, pfBlocker, and Avahi.

Ideas?

**Comment by gnieboer** [comment]
Yes, it appears setup worked.  But there are no devices or entities related to pfSense or the firewall name in the configuration menus.

**Comment by apedance** [comment]
Same issue. pfsense is latest 2.6.0

Config entry '<firewallname>' for pfsense integration not ready yet; Retrying in background

**Comment by chris-a-j** [comment]
I was about to create a new thread but I seem to be having this issue now. Existing setup, pfsense 2.6.0 and latest home assistant core 2022.3.6. 

I think this may have happened just after the Hass core update, but now all entities are unavailable with “retrying in background” in the logs.

**Comment by apedance** [comment]
> I was about to create a new thread but I seem to be having this issue now. Existing setup, pfsense 2.6.0 and latest home assistant core 2022.3.6.
> 
> I think this may have happened just after the Hass core update, but now all entities are unavailable with “retrying in background” in the logs.

I am on Home Assistant 2022.3.5 - issue started happening without an update.

**Comment by tophattwaffle** [comment]
Same issue here. Integration was working well for the longest time. I recently updated Home Assistant which may be the cause, but I've updated HASS many times in the past without this integration breaking.

10 max sessions, PFSense 2.6.0, Home Assistant 2022.3.6

Integration sets up and connects to PFsense fine. HASS then shows the following in the logs:
`2022-03-23 16:55:48 WARNING (MainThread) [homeassistant.config_entries] Config entry 'overlord' for pfsense integration not ready yet; Retrying in background`

PFsense logs > GUI Service shows the following from my HASS IP.
```
Mar 23 16:59:35 | nginx |   | 172.16.22.20 - admin [23/Mar/2022:16:59:35 -0500] "POST /xmlrpc.php HTTP/1.1" 499 0 "-" "Python-xmlrpc/3.9"
Mar 23 16:59:25 | nginx |   | 172.16.22.20 - admin [23/Mar/2022:16:59:25 -0500] "POST /xmlrpc.php HTTP/1.1" 200 249 "-" "Python-xmlrpc/3.9"
Mar 23 16:59:25 | nginx |   | 172.16.22.20 - admin [23/Mar/2022:16:59:25 -0500] "POST /xmlrpc.php HTTP/1.1" 200 274 "-" "Python-xmlrpc/3.9"
Mar 23 16:59:15 | nginx |   | 172.16.22.20 - admin [23/Mar/2022:16:59:15 -0500] "POST /xmlrpc.php HTTP/1.1" 499 0 "-" "Python-xmlrpc/3.9"
```

I have removed the integration through HACS, then reinstalled it - without any luck. I've also fully rebooted the firewall and the machines running Home Assistant.

**Comment by gnieboer** [comment]
@tophattwaffle I notice you have some 499 response codes, my logs are only 200 status codes.  But I have fewer max sessions than you.  Wonder why the difference.

**Comment by chris-a-j** [comment]
Seems to be resolved with latest update?

**Comment by travisghansen** [comment]
Update to hass or this plugin? I’m not (intentionally) ignoring everyone, just a bit busy and still won’t be able to dig in fully for probably another week or so.

**Comment by chris-a-j** [comment]
Sorry I meant the HASS update to 2022.3.7.

I noticed the entities no longer keep going unavailable, but that may just be my setup. 

No urgency on this one of course, as it’s a relatively minor problem in the grand scheme of things!

**Comment by gnieboer** [comment]
I just upgraded the container to 2022.3.7 but the issue appears to be unchanged for me.  I deleted and re-added the integration after the upgrade, no change.  Difference may be that mine is a "new add", the integration was not installed/working previously like @chris-a-j 's.

**Comment by gnieboer** [comment]
I also just upgraded all pfSense packages to latest, tested, and then upgraded from pfSense 2.5.0 to 2.5.2 and tested.  No change in result.

**Comment by aherbjornsen** [comment]
Any progress on this?
I've had this problem for several months now and would really like monitor my pfSense again.

**Comment by travisghansen** [comment]
Doing some maintenance on the project this week and wondering the state of this. Is everyone on this thread only seeing the issue while using an AD user? Have you tried creating a local user and seeing what happens?

**Comment by travisghansen** [comment]
If you disable AD does it work?
